hibernate-core/src/main/java/org/hibernate/engine/jdbc/batch/internal/BatchImpl.java

```diff
@@ -302,13 +303,11 @@ protected void performExecution() {
 				}
 				catch (SQLException e) {
 					abortBatch( e );
-					BATCH_MESSAGE_LOGGER.unableToExecuteBatch( e, sql );
 					throw sqlExceptionHelper.convert( e, "could not execute batch", sql );
 				}
 				catch (RuntimeException re) {
 					abortBatch( re );
-					BATCH_MESSAGE_LOGGER.unableToExecuteBatch( re, sql );
-					throw re;
+					throw new BatchFailedException( "An exception occurred while executing: " + sql, re);
```

Comment:
I don't think we should do this. It looks like this `RuntimeException` might already be a `StaleObjectStateException` or a `BatchFailedException`, in which case re-wrapping it in a(nother) `BatchFailedException` would be really bad.

I think you had it right the first time, @jrenaat.